### ISSUE TITLE
parameter with same name as schema conflicts

### ISSUE THREAD

**Issue by benny-noumena** [ferdikoomen/openapi-typescript-codegen#1977]
We have a parameter and a schema with the same name but they differ in casing.

We noticed that as of 0.24, types for parameters are generated too. This results in a file override on case-insensitive file systems.

Any idea how to solve this? Is there a flag to not generate types for parameters?
What about moving models and parameters in a subdirectory?

Many thanks for advice.

**Comment by markygab** [comment]
I'm seeing something similar too i think. As of 0.24 I'm getting many import errors in the client.

``
src/client/models/WorkflowBase.ts:5:15 - error TS2724: '"./Name"' has no exported member named 'Name'. Did you mean 'name'?``
``
import type { Name } from './Name';
``

Where the file Name.ts  has
``
export type name = string;
``

**Comment by mrlubos** [comment]
@benny-noumena @markygab can you replicate this issue in [our fork](https://github.com/nicolas-chaulet/openapi-typescript-codegen)? It also has a `--useOperationId` flag which you can set to `false` to generate method names based on path (worth trying?)

**Comment by benny-noumena** [comment]
@mrlubos Sadly it still is an issue :(

<img width="374" alt="image" src="https://github.com/ferdikoomen/openapi-typescript-codegen/assets/94522639/9c5ac011-9d53-4252-9443-f8cd91ca9148">

<img width="355" alt="image" src="https://github.com/ferdikoomen/openapi-typescript-codegen/assets/94522639/a2a7da6a-694c-4b98-a463-d2b1604348e6">

<img width="311" alt="image" src="https://github.com/ferdikoomen/openapi-typescript-codegen/assets/94522639/ee4a4f57-ff75-4173-82a8-206b0c87c6e6">

**Comment by mrlubos** [comment]
@benny-noumena are you able to provide spec to replicate this?

**Comment by benny-noumena** [comment]
@mrlubos You mean, like providing an `openapi.yml` fille? The reproduction is simple. If you have a parameter and a schema with the same name, then you end up having to export that differs only in casing. On a case-insensitive file-system this leads to the override of the file. In  the end the `./models/Tenant.ts` and the `./models/tenant.ts` are the same file.

**Comment by mrlubos** [comment]
@benny-noumena do you then import either model or parameter in your code? Is the issue just that the generated client isn't correct?

**Comment by andrew-pledge-io** [comment]
Here's an example schema that demonstrates the problem:

```yml
openapi: 3.0.0
info:
  title: Sample API
  description: Example of conflicting component and parameter schemas
  version: 0.1.9
components:
  schemas:
    Metadata:
      type: string
  parameters:
    metadata:
      in: query
      schema:
        type: string
```

Generating the client and attempting to build it with TypeScript results in the following error:

```
src/index.ts:11:31 - error TS1149: File name 'src/models/metadata.ts' differs from already included file name 'src/models/Metadata.ts' only in casing.
  The file is in the program because:
    Imported via './models/Metadata' from file 'src/index.ts'
    Imported via './models/metadata' from file 'src/index.ts'
    Matched by include pattern 'src' in 'tsconfig.json'
```

Additionally since I'm on a case-insensitive file system only `Metadata.ts` actually exists (assuming `metadata.ts` was overwritten).

**Comment by mrlubos** [comment]
Thank you @andrew-pledge-io. Do you import parameters anywhere in your application code? Any preference for the ideal solution?

**Comment by benny-noumena** [comment]
I don't know. We don't use these parameters for now. However I guess there was a reason why these types were added. My idea was to generate them in a separate directory `parameters`. Like a namespace `parameters`?

**Comment by mrlubos** [comment]
@benny-noumena yeah, it will be a breaking change either way which sucks. I'll try to find out the original motivation for it and go from there. I like the idea of single import module rather than having to go into subdirectories, so maybe prefix would work like we have for schemas ($), would that be okay with you?

**Comment by andrew-pledge-io** [comment]
A prefix would be a good solution. 

Adding a configurable option for the prefix could make it a non-breaking change if the default is an empty prefix.

Detecting path separators in the prefix could also allow the option to be used for generating into a subdirectory? Though that perhaps is complicating things.

**Comment by ufii4** [comment]
Running into the same issue... Either way would be fine, namespace is ideal but I can accept prefix. Important thing is that we have a way to resolve this.
